File: Projects/Project_Cinema Website/frontend/src/services/TicketService.js
import axios from 'axios';

const baseURL = 'http://localhost:8080/api/tickets'; // Replace with your actual API endpoint

class TicketService {

    // GET
    static async getTicketById(ticketId) {
        return (await axios.get(`${baseURL}/getTicket/${ticketId}`)).data;
    }

    static async getTicketByReferenceNo(referenceNo) {
        return (await axios.get(`${baseURL}/getTicketByReferenceNo/${referenceNo}`)).data;
    }

    static async getAllTickets() {
        return (await axios.get(`${baseURL}/getAllTickets`)).data;
    }

    // POST
    static async addTicket(newTicket) {
        return (await axios.post(`${baseURL}/addTicket`, newTicket)).data;
    }

    // PUT (Update by Ticket ID)
    static async updateTicketById(ticketId, updatedTicket) {
        return (await axios.put(`${baseURL}/updateTicketById/${ticketId}`, updatedTicket)).data;
    }

    // PUT (Update by Reference Number)
    static async updateTicketByReferenceNo(referenceNo, updatedTicket) {
        return (await axios.put(`${baseURL}/updateTicketByReferenceNo/${referenceNo}`, updatedTicket)).data;
    }

    // DELETE (Delete by Ticket ID)
    static async deleteTicketById(ticketId) {
        return (await axios.delete(`${baseURL}/deleteTicketById/${ticketId}`)).data;
    }

    // DELETE (Delete by Reference Number)
    static async deleteTicketByReferenceNo(referenceNo) {
        return (await axios.delete(`${baseURL}/deleteTicketByReferenceNo/${referenceNo}`)).data;
    }
}

export default TicketService;

File: Projects/Project_Cinema Website/backend/src/main/java/com/cinema_website/backend/service/TicketService.java
package com.cinema_website.backend.service;

import com.cinema_website.backend.dto.TicketDTO;
import com.cinema_website.backend.repository.TicketRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@SuppressWarnings("all")
@Service
public class TicketService {

    private final TicketRepository ticketRepository;

    @Autowired
    public TicketService(TicketRepository ticketRepository) {
        this.ticketRepository = ticketRepository;
    }

    // GET
    public TicketDTO getTicketById(int ticketId) {
        return ticketRepository.getTicketById(ticketId);
    }

    public String getTicketSummaryById(int ticketId) {
        return ticketRepository.getTicketSummaryById(ticketId);
    }

    public TicketDTO getTicketByReferenceNo(int referenceNo) {
        return ticketRepository.getTicketByReferenceNo(referenceNo);
    }

    public List<TicketDTO> getAllTickets() {
        return ticketRepository.getAllTickets();
    }

    public List<TicketDTO> getTicketsByUserId(int userId) {
        return ticketRepository.getTicketsByUserId(userId);
    }

    // INSERT
    public boolean addTicket(TicketDTO ticketDTO) {
        return ticketRepository.addTicket(ticketDTO);
    }

    // UPDATE
    public boolean updateTicketById(int ticketId, TicketDTO updatedTicketDTO) {
        return ticketRepository.updateTicketById(ticketId, updatedTicketDTO);
    }

    public boolean updateTicketByReferenceNo(int referenceNo, TicketDTO updatedTicketDTO) {
        return ticketRepository.updateTicketByReferenceNo(referenceNo, updatedTicketDTO);
    }

    // DELETE
    public boolean deleteTicketById(int ticketId) {
        return ticketRepository.deleteTicketById(ticketId);
    }

    public boolean deleteTicketByReferenceNo(int referenceNo) {
        return ticketRepository.deleteTicketByReferenceNo(referenceNo);
    }
}
